fdbclient/NativeAPI.actor.cpp

```diff
@@ -2692,8 +2704,22 @@ void debugAddTags(Transaction *tr) {
 
 }
 
+SpanID generateSpanID(bool tracingEnabled) {
+	UID uid = deterministicRandom()->randomUniqueID();
+	if (tracingEnabled) {
+		return SpanID(uid.first(), uid.second());
+	} else {
+		return SpanID(uid.first(), 0);
+	}
+	return uid;
+}
```

Comment:
This function is called for every new transaction that is created. It could be made more efficient by making [UID](https://github.com/apple/foundationdb/blob/b69d0fa68e0d1b7b20e971cfbf3a66d21aa60132/flow/IRandom.h#L64) mutable, but I didn't want to make this change without some input.

If `UID`s were mutable (`SpanID` is typedef'd as `UID`) the generated `UID` could have its second `uint64_t` set to 0 instead of creating a new `SpanID`.